python/ray/tests/test_client.py

```diff
@@ -180,6 +180,10 @@ def test_wait(ray_start_regular_shared):
 def test_remote_functions(ray_start_regular_shared):
     with ray_start_client_server() as ray:
 
+        @ray.remote
+        def sleep_for(t):
+            time.sleep(t)
+
```

Comment:
Could you possibly use a [SignalActor](https://github.com/ray-project/ray/blob/ade684ac0319024f4ffd5efa7c130ef6aeee97d5/python/ray/test_utils.py#L335) here? 

Just to avoid flakiness?